lib/tapioca/dsl/compilers/identity_cache.rb

```diff
@@ -216,7 +216,10 @@ def create_index_fetch_by_methods(field, klass)
           ).void
         end
         def create_aliased_fetch_by_methods(field, klass)
-          type, _ = Helpers::ActiveRecordColumnTypeHelper.new(constant).type_for(field.alias_name.to_s)
+          type, _ = Helpers::ActiveRecordColumnTypeHelper.new(
```

Comment:
> I think we need to update [DelegatedTypes](https://github.com/Shopify/tapioca/blob/main/lib/tapioca/dsl/compilers/active_record_delegated_types.rb#L136) compiler too.

I don't think so since that compiler is just trying to get the `getter_type` but makes it unconditionally nilable later: https://github.com/Shopify/tapioca/blob/main/lib/tapioca/dsl/compilers/active_record_delegated_types.rb#L153

> We also have a separate nilable logic for [Associations](https://github.com/Shopify/tapioca/blob/0007e484ac5d25bb6b7c10bc1c5baa1c81fb7596/lib/tapioca/dsl/compilers/active_record_associations.rb#L190) compiler. What do you think of generalizing this so that it's used for that compiler as well?

I think that needs to be a different option for that compiler.

> Also multiple compilers using this setting makes me think if it's better to not specify the compiler name in the config, is that asking too much of the user? Although most will be happy with the default and won't configure.

Multiple compilers should not be using this option. The fact that we are passing an extra parameter to our internal helper does not mean that the compiler is somehow using an option.

I would like each DSL compiler to be able to accept different options, so I wouldn't really like to tie them together.